src/Polly.Shared/NoOp/NoOpEngine.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+
+namespace Polly.Shared.NoOp
+{
+    internal static partial class NoOpEngine
+    {
+        internal static TResult Implementation<TResult>(Func<CancellationToken, TResult> action,
+            CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
```

Comment:
We could remove `cancellationToken.ThrowIfCancellationRequested();` to make the "noop" concept completely pure.
(and similar elsewhere where same pattern)